### PR TITLE
Add AffectHungerEvent

### PR DESCRIPTION
**Pull request by ktksan**
This event is sent out by the HungerAuthoritySystem for other systems to modify the expected hunger decay.

### CODE REVIEW

**Comment by ktksan** [comment]
The comments changed automatically after Ctrl+alt+L 😅

**Comment by ktksan** [comment]
@skaldarnar @jdrueckert Please take a look at this later.

File: src/main/java/org/terasology/hunger/HungerAuthoritySystem.java [hunk 1]
```diff
@@ -93,6 +94,7 @@
 
     public static final String HUNGER_DAMAGE_ACTION_ID = "Hunger Damage";
     private boolean destroyDrink = false;
+    private float expectedDecay;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/hunger/HungerAuthoritySystem.java]: this is only used in `onPeriodicActionTriggered` and thus can be method-local.

File: src/main/java/org/terasology/hunger/HungerAuthoritySystem.java [hunk 2]
```diff
@@ -112,10 +114,15 @@ public void postBegin() {
     @ReceiveEvent
     public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, EntityRef entityUnused) {
         if (event.getActionId().equals(HUNGER_DAMAGE_ACTION_ID)) {
-            for (EntityRef entity : entityManager.getEntitiesWith(HungerComponent.class, AliveCharacterComponent.class)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(HungerComponent.class,
+                    AliveCharacterComponent.class)) {
                 HungerComponent hunger = entity.getComponent(HungerComponent.class);
-                hunger.lastCalculatedFood = Math.max(0,
-                        hunger.lastCalculatedFood - (healthDecreaseInterval * hunger.foodDecayPerSecond) / 1000);
+                expectedDecay = (healthDecreaseInterval * hunger.foodDecayPerSecond) / 1000;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/hunger/HungerAuthoritySystem.java]:
```suggestion
                final float expectedDecay = (healthDecreaseInterval * hunger.foodDecayPerSecond) / 1000;
```

File: src/main/java/org/terasology/hunger/HungerAuthoritySystem.java [hunk 3]
```diff
@@ -112,10 +114,15 @@ public void postBegin() {
...
+                // Send event to allow for other systems to modify hunger decay.
+                AffectHungerEvent affectHungerEvent = new AffectHungerEvent(expectedDecay);
+                entity.send(affectHungerEvent);
+                expectedDecay = affectHungerEvent.getResultValue();
+                hunger.lastCalculatedFood = Math.max(0, hunger.lastCalculatedFood - expectedDecay);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/hunger/HungerAuthoritySystem.java]: I don't think we need to change the value of the "expected decay". We use it to trigger `AffectHungerEvent` and afterwards are only interested in the result.

```suggestion
                hunger.lastCalculatedFood = Math.max(0, hunger.lastCalculatedFood - affectHungerEvent.getResultValue());
```

**Review by skaldarnar — DISMISSED** [review]

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**